response/response.go

```diff
@@ -97,39 +103,108 @@ func SetDesiredComposedResources(rsp *v1beta1.RunFunctionResponse, dcds map[reso
 }
 
 // Fatal adds a fatal result to the supplied RunFunctionResponse.
-func Fatal(rsp *v1beta1.RunFunctionResponse, err error) {
+// A corresponding event will be created for the Composite Resource.
+func Fatal(rsp *v1beta1.RunFunctionResponse, err error) *ResultOption {
 	if rsp.GetResults() == nil {
 		rsp.Results = make([]*v1beta1.Result, 0, 1)
 	}
-	rsp.Results = append(rsp.GetResults(), &v1beta1.Result{
+	result := &v1beta1.Result{
 		Severity: v1beta1.Severity_SEVERITY_FATAL,
 		Message:  err.Error(),
-	})
+		Target:   v1beta1.Target_TARGET_COMPOSITE,
+	}
+	rsp.Results = append(rsp.GetResults(), result)
+
+	return &ResultOption{
+		result: result,
+	}
 }
 
 // Warning adds a warning result to the supplied RunFunctionResponse.
-func Warning(rsp *v1beta1.RunFunctionResponse, err error) {
+// A corresponding event will be created for the Composite Resource.
+func Warning(rsp *v1beta1.RunFunctionResponse, err error) *ResultOption {
 	if rsp.GetResults() == nil {
 		rsp.Results = make([]*v1beta1.Result, 0, 1)
 	}
-	rsp.Results = append(rsp.GetResults(), &v1beta1.Result{
+	result := &v1beta1.Result{
 		Severity: v1beta1.Severity_SEVERITY_WARNING,
 		Message:  err.Error(),
-	})
+		Target:   v1beta1.Target_TARGET_COMPOSITE,
+	}
+	rsp.Results = append(rsp.GetResults(), result)
+
+	return &ResultOption{
+		result: result,
+	}
 }
 
 // Normal adds a normal result to the supplied RunFunctionResponse.
-func Normal(rsp *v1beta1.RunFunctionResponse, message string) {
+// A corresponding event will be created for the Composite Resource.
+func Normal(rsp *v1beta1.RunFunctionResponse, message string) *ResultOption {
 	if rsp.GetResults() == nil {
 		rsp.Results = make([]*v1beta1.Result, 0, 1)
 	}
-	rsp.Results = append(rsp.GetResults(), &v1beta1.Result{
+	result := &v1beta1.Result{
 		Severity: v1beta1.Severity_SEVERITY_NORMAL,
 		Message:  message,
-	})
+		Target:   v1beta1.Target_TARGET_COMPOSITE,
+	}
+	rsp.Results = append(rsp.GetResults(), result)
+
+	return &ResultOption{
+		result: result,
+	}
 }
 
 // Normalf adds a normal result to the supplied RunFunctionResponse.
-func Normalf(rsp *v1beta1.RunFunctionResponse, format string, a ...any) {
-	Normal(rsp, fmt.Sprintf(format, a...))
+// A corresponding event will be created for the Composite Resource.
+func Normalf(rsp *v1beta1.RunFunctionResponse, format string, a ...any) *ResultOption {
+	return Normal(rsp, fmt.Sprintf(format, a...))
+}
+
+// TargetComposite configures the Composite to receive any events or conditions
+// generated by this result.
+func (o *ResultOption) TargetComposite() *ResultOption {
+	o.result.Target = v1beta1.Target_TARGET_COMPOSITE
+	return o
+}
+
+// TargetBoth configures both the Claim and Composite to receive any events or
+// conditions generated by this result.
+func (o *ResultOption) TargetBoth() *ResultOption {
```

Comment:
```suggestion
func (o *ResultOption) TargetCompositeAndClaim() *ResultOption {
```

This is a bit of a mouthful, but I think it might be best to use the same naming in the protobuf and API.

Reply:
updated to `TargetCompositeAndClaim`